plugins/processors/batch/batch_test.go

```diff
@@ -0,0 +1,69 @@
+package batch
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const batchTag = "?internal_batch_idx"
+
+func MakeBatching(batches uint64) *Batch {
+	return &Batch{
+		BatchTag:   batchTag,
+		NumBatches: batches,
+	}
+}
+
+func MakeXMetrics(count int) []telegraf.Metric {
+	ms := make([]telegraf.Metric, 0, count)
+	for range count {
+		ms = append(ms, testutil.MockMetrics()...)
+	}
+
+	return ms
+}
+
+func requireMetricInBatch(t *testing.T, m telegraf.Metric, batch string) {
+	batchTagValue, ok := m.GetTag(batchTag)
+	require.True(t, ok)
+	require.Equal(t, batch, batchTagValue)
+}
```

Comment:
I would really rather move the code into the tests instead of moving it to functions. The reason is that if you look at a single (maybe failing) test you don't have to navigate around in the file to understand what is going on. I know it duplicates code but I don't think it's worth considering the readability.